lib/server/web.js

```diff
@@ -57,8 +57,8 @@ exports.create = function createServer() {
             token: token
           }
         }, function(err, resp, body) {
-          if (err) {
-            logger.error('auth', err);
+          if (err || resp.statusCode >= 500) {
+            err = err || resp.statusMessage || 'unknown';
             return reply(AppError.oauthError(err));
```

Comment:
So, I'm not sure that during the failure period that resp or resp.statusCode had a value. I don't know why `err` was not truthy though. Hand-waving, but it might need to listen for an `error` event to detect the failure. 

But perhaps just doing 'if !body then return error()' is what is needed.